build-logic/src/main/kotlin/kotlinx/kover/conventions/kover-publishing-conventions.gradle.kts

```diff
@@ -24,19 +21,100 @@ plugins {
 
 interface KoverPublicationExtension {
     val description: Property<String>
-    val fatJar: Property<Boolean>
     val addPublication: Property<Boolean>
+    val localRepository: DirectoryProperty
 }
 
 val extension = extensions.create<KoverPublicationExtension>("koverPublication")
 
 extension.description.convention("")
 extension.addPublication.convention(true)
-extension.fatJar.convention(false)
+extension.localRepository.convention(layout.buildDirectory.dir(".m2"))
+
+internal interface LocalArtifactAttr : Named {
+    companion object {
+        val ATTRIBUTE = Attribute.of(
+            "kotlinx.kover.gradle-plugin",
+            LocalArtifactAttr::class.java
+        )
+    }
+}
+
+val publicationTask: TaskCollection<*> = tasks.matching { task -> task.name == "publishAllPublicationsToLocalRepository" }
+configurations.register("localPublication") {
+    isVisible = false
+    isCanBeResolved = false
+    // this configuration produces modules that can be consumed by other projects
+    isCanBeConsumed = true
+    attributes {
+        attribute(LocalArtifactAttr.ATTRIBUTE, objects.named("local-repository"))
+    }
+
+    outgoing.artifact(extension.localRepository) {
+        builtBy(publicationTask)
+    }
+}
+
+val snapshotRelease = configurations.create("snapshotRelease") {
+    isVisible = true
+    isCanBeResolved = false
+    isCanBeConsumed = false
+}
+
+val externalSnapshots = configurations.create("snapshots") {
+    isVisible = false
+    isCanBeResolved = true
+    // this config consumes modules from OTHER projects, and cannot be consumed by other projects
+    isCanBeConsumed = false
+
+    attributes {
+        attribute(LocalArtifactAttr.ATTRIBUTE, objects.named("local-repository"))
+    }
+
+    extendsFrom(snapshotRelease)
+}
+
+tasks.register<CollectTask>("collectRepository") {
+    dependsOn(publicationTask)
+    dependsOn(externalSnapshots)
+
+    local.convention(extension.localRepository)
+    externals.from(externalSnapshots)
+}
+
+@CacheableTask
+abstract class CollectTask: DefaultTask() {
```

Comment:
I do not see instances and configuration of this task, am I missing something?

Reply:
`tasks.register<CollectTask>("collectRepository") {` above